File: duplicate_row_matrix.cpp
//{ Driver Code Starts
#include <bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++

namespace std {
    template <>
    struct hash<std::vector<int>> {
        size_t operator()(const std::vector<int>& v) const {
            size_t hash_value = 0;
            for (int i : v) {
                hash_value ^= std::hash<int>{}(i) + 0x9e3779b9 + (hash_value << 6) + (hash_value >> 2);
            }
            return hash_value;
        }
    };
}

class Solution {
public:
    std::vector<int> repeatedRows(std::vector<std::vector<int>> &matrix, int M, int N) {
        std::unordered_map<std::vector<int>, std::vector<int>> mp;
        for (int i = 0; i < matrix.size(); i++) {
            std::vector<int> t = matrix[i];
            mp[t].push_back(i);
        }
        std::vector<int> ans;
        for (auto &t : mp) {
            std::vector<int> p = t.second;
            std::sort(p.begin(), p.end());
            if (p.size() > 1) {
                ans.insert(ans.end(), 1+p.begin(), p.end());
            }
        }
        sort(begin(ans),end(ans));
        return ans;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin>>t;
    while(t--) 
    {
        int row, col;
        cin>> row>> col;
        vector<vector<int> > matrix(row); 
        for(int i=0; i<row; i++)
        {
            matrix[i].assign(col, 0);
            for( int j=0; j<col; j++)
            {
                cin>>matrix[i][j];
            }
        }
        
        Solution ob;
        vector<int> ans = ob.repeatedRows(matrix, row, col);


        for (int i = 0; i < ans.size(); ++i)
        {
            cout<<ans[i]<<" ";
        }
        cout<<endl;
    }
    return 0;
}



// } Driver Code Ends

File: duplicate_row_matrix.py
#User function Template for python3

# class Solution:
#     def repeatedRows(self, arr, m ,n):
#         #code here
from typing import List

class Solution:
    def repeatedRows(self, matrix: List[List[int]], M: int, N: int) -> List[int]:
        mp = {}
        for i in range(len(matrix)):
            t = tuple(matrix[i])
            if t in mp:
                mp[t].append(i)
            else:
                mp[t] = [i]
        
        ans = []
        for indices in mp.values():
            sorted(indices)
            if len(indices) > 1:
                ans.extend(indices[1:])
        
        return sorted(ans)
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n, m = list(map(int, input().strip().split()))
        
        arr = []
        for i in range(n):
            nums = list(map(int, input().strip().split()))
            arr.append(nums)
        ob = Solution()
        ans = ob.repeatedRows(arr, n, m)
        
        for x in ans:
            print(x, end=" ")
            
        print()
        tc -= 1

# } Driver Code Ends